tajo-plan/src/main/java/org/apache/tajo/plan/logical/TableSubQueryNode.java

```diff
@@ -108,14 +111,19 @@ public boolean hasTargets() {
   }
 
   @Override
-  public void setTargets(Target[] targets) {
+  public void setTargets(List<Target> targets) {
     this.targets = targets;
     setOutSchema(PlannerUtil.targetToSchema(targets));
   }
 
   @Override
-  public Target[] getTargets() {
-    return targets;
+  public List<Target> getTargets() {
+    if (hasTargets()) {
+      return targets;
```

Comment:
As my understanding of your word, I replace ```return targets;``` with ```Collections.unmodifiableList(targets);``` but several error is occured during unit test.
Am I misunderstood your comment? Can I get some details more about your comment?